week3/average/kirikaza/Average.hs

```diff
@@ -0,0 +1,24 @@
+module Main where
+
+import Control.Monad (forM_)
+import Data.Function (on)
+import qualified Data.Sequence as Seq
+import Data.Sequence ((<|), Seq(..))
+
+main :: IO ()
+main = forM_ [4,2] $ \window -> do
+  print $ movingAvgList window [1,5,3,8,7,9,6]
+
+movingAvgList :: Int -> [Int] -> [Float]
+movingAvgList windowSize values = map avg $ formWindows windowSize values
+
+formWindows :: Int -> [value] -> [Seq value]
+formWindows _ [value] = [Seq.singleton value]
+formWindows size (firstValue:otherValues) =
+  scanl (\ seq e -> e <| Seq.take (size-1) seq)
```

Comment:
`seq` already exists in Prelude, and generally it's not nice to shadow existing names